src/codegen/types.rs

```diff
@@ -14,7 +14,27 @@ pub(crate) fn gen_type_modules(
     prepared: &IndexMap<String, Vec<PreparedType>>,
     config: &Config,
 ) -> proc_macro2::TokenStream {
-    let mut tokens = quote! {};
+    /*
+     Rust types are not first-class values and cannot be assigned to struct fields directly (unlike strings).
+    PhantomData<T> ? std::any::TypeId ? Ditch this idea?
+     */
+    let mut tokens = if config.generate_field_metadata {
+        let field_meta_struct = quote! {
+            #[derive(Debug, Clone, Copy)]
+            pub struct FieldMeta {
+                pub name: &'static str,
+                pub data_type: &'static str,
```

Comment:
suggestion(if-minor): might be useful to add the postgres type as a field as well - thoughts? You should be able to use the `pg_type` fn on `ClorindeType` which gives you the `postgres_types::Type` and I think that has an `impl Display` so it can be converted to a string.

If you add that then - nitpick: I would name them `rust_type` and `pg_type`.